js/exporter/jspdf/export_print_info_calculator.js

```diff
@@ -0,0 +1,121 @@
+import { isDefined, isNumeric } from '../../core/utils/type';
+import { extend } from '../../core/utils/extend';
+
+export const ExportPrintInfoCalculator = {
+    _defaultMinCellHeight: 10,
+
+    _getDefaultOptions: function() {
+        return {
+            margin: {
+                top: 10,
+                right: 10,
+                bottom: 10,
+                left: 10
+            },
+            columnStyles: { },
+            cells: []
+        };
+    },
+
+    calculate: function(options) {
+        options = extend(true, {}, this._getDefaultOptions(), options);
+
+        const {
+            jsPDFDocument,
+            margin,
+            startX,
+            startY,
+            cells,
+            tableWidth,
+            columnStyles
+        } = options;
+
+        const x = startX ?? margin.left;
+        const y = startY ?? margin.top;
+        const width = isNumeric(tableWidth)
+            ? tableWidth
+            : this._calculateTableWidth(jsPDFDocument, x, margin.right);
+
+        this._calculateColumnsStyles(width, columnStyles, cells);
+        this._calculateCellsCoordinates(x, y, cells, columnStyles);
+
+        return extend(true, {}, options, {
+            startX: x,
+            startY: y,
+            tableWidth: width
+        });
+    },
+
+    _calculateTableWidth: function(jsPDFDocument, leftIndent, rightIndent) {
```

Comment:
I think we should postpone automatic tableWidth calculation and work without it for several sprints: it's easy to specify it explicitly and we can declare this value as required at the current stage.